Update concepts->block  to reflect product updates (#13193)
PrefectHQ/prefect: docs/concepts/blocks.md

  - storage
  - secrets
  - configuration
search:
  boost: 2
---

# Blocks

Prefect blocks store configuration and provide an interface for interacting with external systems.

Blocks expose methods that provide pre-built functionality.
For example, blocks can be used to download data from or upload data to an S3 bucket, query data from or write data to a database, or send a message to a Slack channel.

## Overview

Block types are Python classes with a handy UI webform for configuration.
Blocks are instantiation of these classes with specific values.

Configure blocks through Python code or via a form in the UI.
Access blocks for use in Python code.

> Blocks live on in Prefect Cloud or your self-hosted Prefect server instance.

Author: ```suggestion
Block values are stored in Prefect Cloud or your self-hosted Prefect server instance.
```

## Prefect built-in blocks

> Prefect provides a broad range of commonly used, built-in block types.

Reviewer: ```suggestion
Commonly used block types come built-in to Prefect. 
```

Author: ```suggestion
Commonly used block types come built-in with Prefect. 
```

Reviewer: Yeah that's better

> These block types are pre-installed in Prefect Cloud and a self-hosted Prefect server instance.

Reviewer: ```suggestion
```

| [Sendgrid Email](/api-ref/prefect/blocks/notifications/#prefect.blocks.notifications.SendgridEmail)                  | `sendgrid-email`                                         | Enables sending notifications via Sendgrid email.                                         |
| [Slack Webhook](/api-ref/prefect/blocks/notifications/#prefect.blocks.notifications.SlackWebhook)                    | `slack-webhook`                                          | Enables sending notifications via a provided Slack webhook.                      |
| [SMB](/concepts/filesystems/#smb)                                                                                    | `smb`                                                    | Store data as a file on a SMB share.                                                     |
| [String](/api-ref/prefect/blocks/system/#prefect.blocks.system.String)                                               | `string`                                                 | A block that represents a string.                                         |

> | [Twilio SMS](/api-ref/prefect/blocks/notifications/#prefect.blocks.notifications.TwilioSMS)                          | `twilio-sms`                                             | Enables sending notifications via Twilio SMS.                                                       |

Reviewer: ```suggestion
| [Twilio SMS](/api-ref/prefect/blocks/notifications/#prefect.blocks.notifications.TwilioSMS)                          | `twilio-sms`                                             | Sends notifications via Twilio SMS.                                                       |
```

```

> !!! tip "Sharing Blocks"

Author: ```suggestion
```

>     If using [Prefect Cloud](/cloud/), blocks can be loaded by workspace collaborators.

Author: ```suggestion
```

```

> !!! tip "Sharing Blocks"

Reviewer: I think we can get rid of this tip.

### Nested blocks

> Blocks are composable, meaning that blocks can be used within other blocks.

Reviewer: ```suggestion
Blocks are composable - a block can be used within other blocks.
```
